src/halfpipe/tui/templates/feature_template.py

```diff
@@ -0,0 +1,456 @@
+        The specific feature field in the settings.
+    type : str
+        The type of the feature.
+    file_panel_class : type
+        The class used for the file panel within this widget.
+    feature_dict : dict
+        A dictionary containing feature-specific settings and values.
+    setting_dict : dict
+        A dictionary containing general settings and configuration values.
+    event_file_pattern_counter : int
+        A counter for file patterns.
+    tagvals : list
+        A list of available tags for selection.
+    bandpass_filter_low_key : str
+        The key for the low-pass filter setting.
+    bandpass_filter_high_key : str
+        The key for the high-pass filter setting.
+    images_to_use : dict
+        A dictionary specifying which images to use, keyed by task.
+    confounds_options : dict
+        Available options for confounds removal, with their descriptions and default states.
+    preprocessing_panel : Vertical
+        A panel containing pre-processing options such as smoothing, mean scaling, and temporal filtering.
+    tasks_to_use_selection_panel : SelectionList
+        A panel containing the selection list of images to use.
+    tag_panel : SelectionList
+        A panel containing the selection list of tags.
+
+    Methods
+    -------
+    __init__(this_user_selection_dict, id=None, classes=None)
+        Initializes the widget with user selections and settings.
+    compose()
+        Composes the user interface elements within the widget.
+    on_file_panel_file_item_is_deleted(message)
+        Handles the event when a file item is deleted from the file panel.
+    on_file_panel_changed(message)
+        Handles the event when the file panel changes, updating the tag selection.
+    """
+
+    entity: str = ""
+    filters: dict = {"datatype": "", "suffix": ""}
+    featurefield: str = ""
+    type: str = ""
+    file_panel_class = FilePanelTemplate
+
+    def __init__(self, this_user_selection_dict: dict, id: str | None = None, classes: str | None = None) -> None:
+        """At the beginning there is a bunch of 'if not in'. If a new widget is created the pass
+        this_user_selection_dict is empty and the nested keys need some initialization. On the other
+        hand, if a new widget is created automatically on spec file load then this dictionary is not empty and these
+        values are then used for the various widgets within this widget.
+        """
+        super().__init__(id=id, classes=classes)
+        self.feature_dict = this_user_selection_dict["features"]
+        self.setting_dict = this_user_selection_dict["settings"]
+        self.event_file_pattern_counter = 0
+
+        self.temp_bandpass_filter_selection: dict
+        self.feature_dict.setdefault("contrasts", [])
+        self.feature_dict.setdefault("type", self.type)
+
+        self.bandpass_filter_default_switch_value = True
+        if self.type in ["reho", "falff", "atlas_based_connectivity"]:
```

Comment:
I think the subclass-specific code should be in the subclasses such as ReHo, etc. By having this code here in the base class, we might miss important places that need to be modified if we ever update a subclass. It also makes it harder to see what the subclass does from just reading the subclass code.

Reply:
should be now ok after introducing standards.py